### PR TITLE
fix(#23) Applier checks if the returned test classes should be really run

### PR DESCRIPTION
**Pull request by MatousJobanek**
It checks if it is in the list (`TestToRun`) that was previously retrieved when the classpath was scanned.

### CODE REVIEW

File: surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/info/TestNgProviderInfo.java [hunk 1]
```diff
@@ -12,6 +12,7 @@ public TestNgProviderInfo(ProviderParametersParser paramParser) {
 
     public String getProviderClassName() {
         return "org.apache.maven.surefire.testng.TestNGProvider";
+
```
Comment (bartoszmajsak) [inline surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/info/TestNgProviderInfo.java]: No need for that I guess :)

**Comment by bartoszmajsak** [comment]
What is the corner case demonstrating this issue?

File: surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/info/TestNgProviderInfo.java (hunk 1 above)
Comment (MatousJobanek) [inline surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/info/TestNgProviderInfo.java]: Ah, yes - I needed some change to verify the fix :-D

**Comment by MatousJobanek** [comment]
I changed the `TestNgProviderInfo.java` class and the `changed-ftest` failed as it returns also this class.

**Pull request merged**